product/src/main/java/com/yas/product/service/ProductService.java

```diff
@@ -769,9 +750,15 @@ public List<ProductThumbnailGetVm> getFeaturedProductsById(List<Long> productIds
     }
 
     public ProductFeatureGetVm getListFeaturedProducts(int pageNo, int pageSize) {
-        Pageable pageable = PageRequest.of(pageNo, pageSize);
+
         List<ProductThumbnailGetVm> productThumbnailVms = new ArrayList<>();
-        Page<Product> productPage = productRepository.getFeaturedProduct(pageable);
+        List<Long> productIds = inventoryService.getProductIdsAddedWarehouse();
```

Comment:
The current approach of fetching all available products in real-time should work. However, I don't like this approach so much as it impacts the responsibility if the number of products or requests is high. Please consider caching or an event-driven approach.